docs/core/integration-quality-scale/rules/iqs022.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "IQS022 - Entities event setup"
+---
+
+IQS022 is as follows:
+
+Entities event setup
+
+## Reasoning
+
+Entities can be setup to listen directly to events coming from the library.
+In order to do this correctly, the entities should subscribe to the events in `async_added_to_hass`.
+And to avoid memory leaks, the entities should unsubscribe from the events in `async_will_remove_from_hass`.
```

Comment:
```suggestion
In order to do this correctly, the entities should subscribe and register the update callback in the entity method `async_added_to_hass`. This entity method is called after the entity has been registered by the entity platform helper and the entity will now have all its interfaces available to call, such as `self.hass` and `self.async_write_ha_state`. Registering an update callback before this stage will cause errors if the callback eg. tries to access `self.hass` or write a state update.
To avoid memory leaks, the entities should unsubscribe from the events, ie. unregister the update callback, in the entity method `async_will_remove_from_hass`.
```